flang/include/flang/Optimizer/Support/InternalNames.h

```diff
@@ -133,6 +133,10 @@ struct NameUniquer {
   /// Check whether the name should be re-mangle with external ABI convention.
   static bool needExternalNameMangling(llvm::StringRef uniquedName);
 
+  /// Does \p uniquedName belong to module \p moduleName ?
+  static bool belongsToModule(llvm::StringRef uniquedName,
```

Comment:
From this description I wonder if there are restrictions on `uniqueName`. Is it meant to work for global variables, procedures, and types? A submodule?